yb-voyager/cmd/analyzeSchema.go

```diff
@@ -361,6 +359,43 @@ func checkGist(sqlInfoArr []sqlInfo, fpath string) {
 	}
 }
 
+func checkTableDDLs(sqlInfoArr []sqlInfo, fpath string) {
+	for _, sqlStmtInfo := range sqlInfoArr {
+		parseTree, err := pg_query.Parse(sqlStmtInfo.stmt)
+		if err != nil {
+			utils.ErrExit("failed to parse the stmt %v: %v", sqlStmtInfo.stmt, err)
+		}
+
+		createTableNode, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateStmt)
+		if ok {
+			schemaName := createTableNode.CreateStmt.Relation.Schemaname
+			tableName := createTableNode.CreateStmt.Relation.Relname
+			columns := createTableNode.CreateStmt.TableElts
+			var generatedColumns []string
+			for _, column := range columns {
+				//In case CREATE DDL has PRIMARY KEY(column_name) - it will be included in columns but won't have columnDef as its a constraint
+				if column.GetColumnDef() != nil {
+					constraints := column.GetColumnDef().Constraints
+					for _, constraint := range constraints {
+						if constraint.GetConstraint().Contype == pg_query.ConstrType_CONSTR_GENERATED {
+							generatedColumns = append(generatedColumns, column.GetColumnDef().Colname)
+						}
+					}
+				}
+			}
+			fullyQualifiedName := tableName
+			if schemaName != "" {
+				fullyQualifiedName = schemaName + "." + tableName
+			}
+			if len(generatedColumns) > 0 {
+				summaryMap["TABLE"].invalidCount[sqlStmtInfo.objName] = true
+				reportCase(fpath, STORED_GENERATED_COLUMN_ISSUE_REASON+fmt.Sprintf(" Generated Columns: (%s)", strings.Join(generatedColumns, ",")),
+					"https://github.com/yugabyte/yugabyte-db/issues/10695", "Using Triggers to update the generated columns is one way to work around this issue, refer link for more details: <LINK_DOC>", "TABLE", fullyQualifiedName, sqlStmtInfo.formattedStmt)
+			}
+		}
+	}
+}
+
```

Comment:
hi @priyanshi-yb, regarding pg_query and parsing related code if we can encapsulate them and abstract out the functionalities, hiding these internal things like - `parseTree.Stmts[0].Stmt.Node`, `createTableNode.CreateStmt.Relation.Schemaname` etc...

let me know what do you think about having a separate class kind of thing for this.

Reply:
That can be reused for the old code of parser in `exportSchema` and everywhere in analyzeSchema also....

Reply:
I thought about it and realized that our usage in the export schema is minimal and restricted to Create table only, but in analyze we will have different kinds of node types like alterTable, createIndex, createConv..etcc.. So I think it won't be of much use if we create separate class as we will be trying to fetch smaller details from these parsed nodes for different cases. and encapsulating each of them will not be very useful.
What do you think?

Reply:
ok, just think about it, may not need to do that right away.. 
even for some of the parts, abstraction might make the code more readable...

Regarding different kind of node types being used - will there be anything common in them, like getting the similar info for each node etc... If so, then you can think about this.

My point is to avoid the low level details spilled all over the code and keeping all that in a single place.

Reply:
Hmm will think about it later once I have all these reporting issues in and have bandwidth.

Reply:
I think I agree with the general idea of abstracting it out, but let's also be wary of prematurely designing. 

I think the simplest thing I can think of at this moment is: 
Let's parse once, and then hand off to different functions to find issues. General template could look like: 
```
checkStmtsUsingParser():
    parseTree, err := pg_query.Parse(sqlStmtInfo.stmt)
    createTableNode, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateStmt)
    if ok {
        reportGeneratedAlwaysStoredColumns(createTableNode)
        reportInheritedTables(createTableNode)
        ...
       }
    createIndexNode, ok :=   ...
    if ok {
        reportAllUnsupportedIndexes(createIndexNode)...
    }
```

Reply:
Yes, makes sense this is something I was also planning to follow to have to parse a type of node at once only and then report the issue separately. Will follow this further, made the changes in the PR as well